pre_commit_hooks/make/phony-targets.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+set -e
+
+# Make environment variables working in OSX GUI apps such as Github Desktop
+# https://stackoverflow.com/q/135688/483528
+export PATH=$PATH:/usr/local/bin
+
+EXITCODE=0
+
+validate_phony_targets() {
+    local source=$1
+    local targets=$(sed -n 's#^.PHONY:\(.*\)#\1#p' <<< $source)
+
+    for target in $targets; do
+        # The q command for setting exit codes if a pattern doesn't match,
+        # isn't supported in OSX / BSD sed.
+        # We check the exact match as a workaround instead.
+        # If the pattern doesn't match that means that the target doesn't exist.
+        if test -z "$(sed -ne "s#^\(${target}:\).*#\1#p" <<< $source)"; then
```

Comment:
`grep` should do the trick out of the box and might be more readable (stop after first match, do not output match, extended regex (should work without, but be clear here))

```suggestion
        if grep -m 1 -q -E  "^${target}:" <<<"${source}" ; then
```

Reply:
I've used grep in the first place but the problem is the missing compatibility with OSX and BSDs so I switched to sed instead.

Reply:
what are those compatibility issues? grep is standard ;)

Reply:
Yip, grep is standard. However, the grep shipped with OSX differs from GNU grep which can be installed through homebrew. Per default, OSX grep won't allow for PERL regex to be used, options and parameters are different, etc. For example, you suggestion won't work in OSX :)

Reply:
I would also prefer grep over sed but since the current implementation that is based on sed works on both, BSD and Unix, I would suggest that we just stick to it for now.